src/dsl.jl

```diff
@@ -516,155 +450,212 @@ end
+
+# Returns the `default_reaction_metadata` output. Technically Catalyst's code could have been made
+# more generic to account for other default reaction metadata. Practically, this will likely
+# be the only relevant reaction metadata to have a default value via the DSL. If another becomes
+# relevant, the code can be rewritten to take this into account.
+# Checks if the `@default_noise_scaling` option is used. If so, uses it as the default value of
+# the `default_noise_scaling` reaction metadata, otherwise, returns an empty vector.
+function read_default_noise_scaling_option(options)
     if haskey(options, :default_noise_scaling)
-        if (length(options[:default_noise_scaling].args) != 3) # Because of how expressions are, 3 is used.
-            error("@default_noise_scaling should only have a single input, this appears not to be the case: \"$(options[:default_noise_scaling])\"")
+        if (length(options[:default_noise_scaling].args) != 3)
+            error("@default_noise_scaling should only have a single expression as its input, this appears not to be the case: \"$(options[:default_noise_scaling])\"")
         end
-        push!(default_reaction_metadata.args,
-            :(:noise_scaling => $(options[:default_noise_scaling].args[3])))
+        return :([:noise_scaling => $(options[:default_noise_scaling].args[3])])
     end
+    return :([])
 end
 
-# When compound species are declared using the "@compound begin ... end" option, get a list of the compound species, and also the expression that crates them.
-function read_compound_options(opts)
-    # If the compound option is used retrieve a list of compound species (need to be added to the reaction system's species), and the option that creates them (used to declare them as compounds at the end).
-    if haskey(opts, :compounds)
-        compound_expr = opts[:compounds]
-        # Find compound species names, and append the independent variable.
-        compound_species = [find_varinfo_in_declaration(arg.args[2])[1]
-                            for arg in compound_expr.args[3].args]
+# When compound species are declared using the "@compound begin ... end" option, get a list
+# of the compound species, and also the expression that crates them.
+function read_compound_options(options)
+    # If the compound option is used, retrieve a list of compound species and  the option line
+    # that creates them (used to declare them as compounds at the end).
+    if haskey(options, :compounds)
+        cmpexpr_init = options[:compounds]
+        cmps_declared = [find_varinfo_in_declaration(arg.args[2])[1]
+                            for arg in cmpexpr_init.args[3].args]
     else  # If option is not used, return empty vectors and expressions.
-        compound_expr = :()
-        compound_species = Union{Symbol, Expr}[]
+        cmpexpr_init = :()
+        cmps_declared = Union{Symbol, Expr}[]
     end
-    return compound_expr, compound_species
+    return cmpexpr_init, cmps_declared
```

Comment:
shortened compound to cmp in line with other shortenings, and to avoid having to break certain lines into two lines in some places.